kiali_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/kiali/kiali/config"
+)
+
+func TestValidateWebRoot(t *testing.T) {
+	// create a base config that we know is valid
+	conf := config.NewConfig()
+	conf.Server.StaticContentRootDirectory = "."
+	conf.Auth.Strategy = "anonymous"
+
+	// now test some web roots, both valid ones and invalid ones
+	validWebRoots := []string{
+		"/",
+		"/kiali",
+		"/abc/clustername/api/v1/namespaces/istio-system/services/kiali:80/proxy/kiali",
+		"/a/0/-/./_/~/!/$/&/'/(/)/*/+/,/;/=/:/@/%aa",
+		"/kiali0-._~!$&'()*+,;=:@%aa",
+	}
+	invalidWebRoots := []string{
```

Comment:
not sure, but should we disallow `x/../y` from a security perspective ? (the `..`)

Reply:
I am not sure either, but "just in case", see my latest commit where that is now disallowed and you get an explicit error message if web root is trying to use that "..".